patches/transport-timeout-interval.patch

```diff
@@ -0,0 +1,88 @@
+
+
+From: nobody <nobody@nowhere>
+
+
+---
+ src/BasicTransport.cc |    5 +++++
+ src/OptionParser.cc   |    5 +++++
+ src/OptionParser.h    |   13 ++++++++++++-
+ 3 files changed, 22 insertions(+), 1 deletion(-)
+
+diff --git a/src/BasicTransport.cc b/src/BasicTransport.cc
+index 5253669a..2137e07c 100644
+--- a/src/BasicTransport.cc
++++ b/src/BasicTransport.cc
+@@ -15,6 +15,7 @@
+ #include <algorithm>
+ 
+ #include "BasicTransport.h"
++#include "OptionParser.h"
+ #include "Service.h"
+ #include "TimeTrace.h"
+ #include "WorkerManager.h"
+@@ -110,6 +111,10 @@ BasicTransport::BasicTransport(Context* context, const ServiceLocator* locator,
+     timerInterval = Cycles::fromMicroseconds(2000);
+     nextTimeoutCheck = Cycles::rdtsc() + timerInterval;
+ 
++    if (context->options != NULL) {
++        timeoutIntervals = context->options->getTimeoutIntervals();
```

Comment:
`if (context->options->getTimeoutIntervals() != 0)` should probably go here. If somebody doesn't specify this option, then it defaults to `0`. If we use the value of `0` and overwrite the RAMCloud hardcoded default then that's bad and I think that will cause a crash later due to some assertions RAMCloud makes. At any rate, it seems best to use the hardcoded default if `getTimeoutIntervals()` returns `0`. Your comments in `OptionParser.cc` about the default being `40` also indicate that this should be done.

Reply:
Modified here. Also set default in OptionParser.cc to zero for consistency w other arguments. I verified by hand that, when timeoutIntervals argument isn't specified, we DO use the default value of 40 due to default constructor value population in BasicTransport.cc. We NEVER actually want timeoutIntervals to be zero in BasicTransport.cc, and anyhow, there's an assert line further down in this same file that effectively requires timeoutIntervals to always be higher than 2 * pingIntervals (which is hardcoded to 3, hence timeoutIntervals must always be higher than 6)